cli/src/build-settings.ts

```diff
@@ -13,35 +14,46 @@ function isStringDict(x: JSON.Object): x is Dict<string | null> {
 
 export default class BuildSettings {
   private rustc: string;
+  private nodeVersion: string | null;
   private env: Dict<string | null>;
 
-  constructor(rustc: string, env: Dict<string | null>) {
+  constructor(rustc: string, nodeVersion: string | null, env: Dict<string | null>) {
     this.rustc = rustc;
+    this.nodeVersion = nodeVersion
     this.env = env;
   }
 
   match(other: BuildSettings) {
+    if (other.nodeVersion !== this.nodeVersion) return false;
     return Object.keys(this.env).every(key => {
       return (!this.env[key] && !other.env[key]) ||
              (this.env[key] === other.env[key]);
     });
   }
 
-  static current(toolchain: rust.Toolchain) {
-    const spawnResult = rust.spawnSync("rustc", ["--version"], toolchain);
+  static getNodeVersion(): string {
+    const nodeVersionResult = child_process.spawnSync("node", ["--version"]);
```

Comment:
Can you check `process.version` or `process.versions`?

Reply:
Are you talking about like the following:

```js
const nodeVersionsResult =
  child_process.spawnSync("node", ["-e 'console.log(process.versions)'"]);
```